libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/HttpHelper.cs

```diff
@@ -33,8 +34,19 @@ public static async Task<Activity> ReadRequestAsync(HttpRequestMessage request,
                 throw new ArgumentNullException(nameof(request));
             }
 
-            var activity = await request.Content.ReadAsAsync<Activity>(BotMessageMediaTypeFormatters, cancellationToken).ConfigureAwait(false);
-            return activity;
+            try
+            {
+                var activity = await request.Content.ReadAsAsync<Activity>(BotMessageMediaTypeFormatters, cancellationToken).ConfigureAwait(false);
+                return activity;
+            }
+            catch (UnsupportedMediaTypeException)
+            {
+                return null;
+            }
+            catch (JsonException)
+            {
+                return null;
```

Comment:
Logging

Reply:
Ditto point about logging - look at the HTTP traffic for the result. Bad Request is exactly that. Its not a server error - at the application level the client should log this. This makes HTTP logs more meaningful.